### ISSUE TITLE
Variables on the stack are sometimes not correctly located

### ISSUE THREAD

**Issue by nikolashn** [nikolashn/basque#3]
Example of code compiled erroneously:
```py
{
    i64 x = 5;
    i64 y = 3;
    i64 a = x * y;
    # Writes "5" instead of "15"
    write a; "\n";
}
```
Disassembly of the code generated by this example gives the following (comments added by me):
```nasm
[...]
0000010F  B805000000        mov eax,0x5
00000114  50                push rax
00000115  B803000000        mov eax,0x3
0000011A  50                push rax
0000011B  488B442410        mov rax,[rsp+0x10] ; (incorrect location) x
00000120  488B4C2408        mov rcx,[rsp+0x8] ; (incorrect location) y
00000125  480FAFC1          imul rax,rcx
00000129  50                push rax
0000012A  488B0424          mov rax,[rsp] ; (correct location) a
0000012E  E8CDFEFFFF        call 0x0 ; (internal function) U64ToStr
[...]
```

**Comment by nikolashn** [comment]
Bug is in parser/funcvar.h:277 (in ba_PVarDef), due to ctr->currScope->dataSize being expanded before the assigned statement is evaluated. Will be fixed in next (or so) commit.

**Comment by nikolashn** [comment]
Fixed by 0fe581264cf56520f9aa861c91be44f1ae877c65
